### ISSUE TITLE
Unknown option "--enable-libsvtvp9".

### ISSUE THREAD

**Issue by kusaanko** [rdp/ffmpeg-windows-build-helpers#560]
When I build with `./cross_compile_ffmpeg.sh --build-ffmpeg-static=y --build-ffmpeg-shared=y --enable-gpl=n --compiler-flavors=win64 --build-intel-qsv=y --build-amd-amf=y`, an error was occured.

```
config options --pkg-config=pkg-config --pkg-config-flags=--static --extra-version=ffmpeg-windows-build-helpers --enable-version3 --disable-debug --disable-w32threads --arch=x86_64 --target-os=mingw32 --cross-prefix=/home/runner/work/FFmpeg-Auto-Build/FFmpeg-Auto-Build/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/bin/x86_64-w64-mingw32- --enable-libcaca --enable-gray --enable-libtesseract --enable-fontconfig --enable-gmp --enable-gnutls --enable-libass --enable-libbluray --enable-libbs2b --enable-libflite --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libilbc --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopus --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvo-amrwbenc --enable-libvorbis --enable-libwebp --enable-libzimg --enable-libzvbi --enable-libmysofa --enable-libopenjpeg --enable-libopenh264 --enable-liblensfun --enable-libvmaf --enable-libsrt --enable-demuxer=dash --enable-libxml2 --enable-opengl --enable-libdav1d --enable-cuda-llvm --enable-libsvthevc --enable-libsvtav1 --enable-libsvtvp9 --enable-libaom --enable-libvpx --enable-nvenc --enable-nvdec --extra-libs=-lharfbuzz --extra-libs=-lm --extra-libs=-lpthread --extra-cflags=-DLIBTWOLAME_STATIC --extra-cflags=-DMODPLUG_STATIC --extra-cflags=-DCACA_STATIC --enable-amf --enable-libmfx --enable-avresample --extra-cflags=-mtune=generic --extra-cflags=-O3 --enable-static --disable-shared --prefix=/home/runner/work/FFmpeg-Auto-Build/FFmpeg-Auto-Build/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32 ./configure
Unknown option "--enable-libsvtvp9".
See ./configure --help for available options.
failed configure ffmpeg_git_lgpl
##[error]Process completed with exit code 1.
```

But I can build with `./cross_compile_ffmpeg.sh --build-ffmpeg-static=y --build-ffmpeg-shared=y --enable-gpl=n --compiler-flavors=win64 --build-intel-qsv=y --build-amd-amf=y --ffmpeg-git-checkout-version=n4.4`.

How do I fix it?

**Comment by nanake** [comment]
ping @jojje

**Comment by nanake** [comment]
```
error: can't open patch '../SVT-HEVC_git/ffmpeg_plugin/0001-lavc-svt_hevc-add-libsvt-hevc-encoder-wrapper.patch': No such file or directory
error: can't open patch '../SVT-VP9_git/ffmpeg_plugin/master-0001-Add-ability-for-ffmpeg-to-run-svt-vp9.patch': No such file or directory
configuring ffmpeg-source
...
Unknown option "--enable-libsvthevc".
See ./configure --help for available options.
failed configure ffmpeg-source
##[error]Process completed with exit code 1.
```

build busted since commit d2342d6

@rdp maybe you need to set up github actions to check pull request.

**Comment by jojje** [comment]
Hi, thanks for reporting this.
The commit _did_ work at the time it was merged. Since then SVT VP9 has drifted from ffmpeg:master to such an extent that the patch they provide no longer applies.

The issue I found right now when trying to build the project with the flags provided by OP was this:
```diff
+ git apply ../SVT-HEVC_git/ffmpeg_plugin/0001-lavc-svt_hevc-add-libsvt-hevc-encoder-wrapper.patch
+ git apply /ffmpeg-windows-build-helpers/patches/SVT-HEVC-0002-doc-Add-libsvt_hevc-encoder-docs.patch
+ git apply ../SVT-VP9_git/ffmpeg_plugin/master-0001-Add-ability-for-ffmpeg-to-run-svt-vp9.patch
error: patch failed: libavformat/matroskaenc.c:2330
error: libavformat/matroskaenc.c: patch does not apply
```

This problem has already been identified upstream: 
https://github.com/OpenVisualCloud/SVT-VP9/issues/145

For now you can just comment out line 2313 (`git apply "../SVT-VP9_git/ffmpeg_plugin`...) and line 2340 (`config_options+=" --enable-libsvtvp9"`) in `cross_compile_ffmpeg.sh` to get the builds working again.

I'm somewhat hesitant to create a local rebase of SVT's VP9 patch, since it's likely the upstream project team will rebase themselves and merge at any time. If I provide a local patch (of their patch), then builds using this project will fail again as soon as upstream has rebased the problematic patch.

What do you guys think? What's the best way to resolve this issue short and long term?
What's the best way in general to catch the issue of dependent project drifts?

Personally I think having a daily automated build would be the best option, as it would be a canary showing everyone whether or not the project can produce a build at any given date. However, that will likely incur some cost which someone needs to pay for.

**Comment by jojje** [comment]
The PR just disables VP9 for now, as mentioned above. That will at least unblock those of you who don't need VP9 and are suffering from this.

**Comment by jojje** [comment]
Btw @nanake your suggestion of checking pull requests would not have helped for cases like this, because such tests only assert that a PR works at the specific time the test was run. This project relies on pulling a gazillion projects from their respective master branches, which means there is zero guarantee that a build that works one second will work again the next. As was the case with this particular issue, which you incorrectly asserted was caused by the commit merged into this project.

The only way around drift issues like this is to pin dependencies and ffmpeg alike.

If you're looking for guarantees that builds don't break, then this is not the project you are looking for. To achieve it with this project you'll just have to try get a combination of dependencies that work with whatever version of ffmpeg is being used, get a successful build out of that, and then _disable_ all git-pulls / source downloads in the script. That will give you reproducible build behavior.

**Comment by nanake** [comment]
Two scenarios here, (a) bitrotted patch and (b) apply external patch.

[My issue](https://github.com/rdp/ffmpeg-windows-build-helpers/issues/560#issuecomment-822958721) from (b).

People use this repo on their github actions to auto-build ffmpeg and it break ever since commit d2342d6. the log i provided above was from `github actions`.

`No such file or directory`

This means applying an external patch is totally a bad idea. You'll inevitably break their CI.

**Comment by nanake** [comment]
> The commit did work at the time it was merged. Since then SVT VP9 has drifted from ffmpeg:master to such an extent that the patch they provide no longer applies.

My takeaway: people should refrain from patching over high entropy project (ffmpeg:master).

Use `opt-in`, not [opt-out](url) if you want to.

**Comment by bootrino** [comment]
Am I right in reading that SVT VP9 for ffmpeg is essentially unmaintained/dead?

**Comment by rdp** [comment]
Looks like it, you could try it and let me know if it still works I could make it opt in...

**Comment by bootrino** [comment]
So is it possible to use this with ffmpeg?  If yes, can anyone suggest how? thanks!

**Comment by jojje** [comment]
Yes. I found the SVT quality sub-par and have since just settled on HEVC (x265). Best overall compromises for my needs; quality/file size, compatibility and encoding time.

Since SVT is still not merged into upstream ffmpeg, I can submit a patch to remove the that stuff from this project. Maintaining something I don't use myself won't happen unfortunately, and having broken code in this project serves no-one.

**Comment by rdp** [comment]
Thanks for your help over time :)

**Comment by stickman561** [comment]
Dang, didn't realize SVT-VP9 would introduce so much headache when I added it. Sorry about that. Guess it's not as well-maintained upstream as SVT-AV1, and I've been too busy with Grad School to watch and maintain.

**Comment by rdp** [comment]
I think it's disabled by default.  If "she's dead, Jim" then yeah we could
remove it :)
